probe-rs/src/debug/source_instructions.rs

```diff
@@ -230,15 +300,22 @@ impl Debug for InstructionSequence<'_> {
             self.address_range.start, self.address_range.end
         )?;
         for instruction_location in &self.instructions {
-            writeln!(f, "\t{instruction_location:?}")?;
+            writeln!(
+                f,
+                "\t{instruction_location:?} - {}",
+                self.debug_info
+                    .get_path(&self.program_unit.unit, instruction_location.file_index)
+                    .map(|file_path| file_path.to_string_lossy().to_string())
+                    .unwrap_or("<unknown file>".to_string())
```

Comment:
I know this is a non-issue but I think I'd prefer trying `Cow`s here over the two `to_string`s. I'd appreciate not introducing allocations where we don't need to, regardless of where we do it.

Reply:
@bugadani I can't return to `Cow` from the closure. Do you have a suggestion?

Reply:
This works, but I'll let you decide if it's an improvement or not.

```rust
            let path = self
                .debug_info
                .get_path(&self.program_unit.unit, instruction_location.file_index);
            let file = path
                .as_ref()
                .map(|file_path| file_path.to_string_lossy())
                .unwrap_or(std::borrow::Cow::Borrowed("<unknown file>"));
            writeln!(f, "\t{instruction_location:?} - {file}")?;
```

Reply:
Is this not just moving an alloc to the `let path`?

Reply:
No, just binds it to a variable, and takes a reference. In the end it's really trading an allocation for an `as_ref()` call. But the end result is three more lines of code, which isn't particularly nice, I admit.

Reply:
Cool. I'm ok to leave it as is, if you are.